pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -61,10 +66,37 @@ func (az *Cloud) GetLoadBalancer(clusterName string, service *v1.Service) (statu
 	}, true, nil
 }
 
+func (az *Cloud) getPublicIPName(clusterName string, service *v1.Service) (string, error) {
+	loadBalancerIP := service.Spec.LoadBalancerIP
+	if len(loadBalancerIP) == 0 {
+		return fmt.Sprintf("%s-%s", clusterName, cloudprovider.GetLoadBalancerName(service)), nil
+	}
+
+	list, err := az.PublicIPAddressesClient.List(az.ResourceGroup)
```

Comment:
I forgot how much this is oriented around "the" resource group. This returns the public ip name, and all of the code that looks at the public ip looks for it in the current resource group. I think you have to revert the last change or make a larger change to refactor `getPublicIPName` to `getPublicIP`.